synapse/http/server.py

```diff
@@ -46,6 +46,8 @@
 from twisted.internet.defer import CancelledError
 from twisted.python import failure
 from twisted.web import resource
+from twisted.web.pages import notFound
```

Comment:
For old deps I think we need:

```suggestion
try:
    from twisted.web.pages import notFound
except ImportError:
    from twisted.web.resource import NoResource as notFound
```